sap-spring-properties/src/main/java/com/sap/cloud/environment/servicebinding/SapSpringPropertiesServiceBindingAccessor.java

```diff
@@ -0,0 +1,104 @@
+    @Nonnull
+    private static final Logger logger = LoggerFactory.getLogger(SapSpringPropertiesServiceBindingAccessor.class);
+
+    @Nonnull
+    @Override
+    public List<ServiceBinding> getServiceBindings()
+            throws ServiceBindingAccessException {
+        if (SapServiceBindingsPropertiesAccessor.getServiceBindingsProperties().isEmpty()) {
+            logger.info("No service bindings found in properties.");
+            return Collections.emptyList();
+        }
+
+        List<ServiceBinding> serviceBindings = fetchServiceBindings();
+
+        serviceBindings.forEach(binding ->
+                logger.info("Service binding {} found in properties.", binding.getServiceName().get()) //NOSONAR
+        );
+        return serviceBindings;
+    }
+
+    @Nonnull
+    private List<ServiceBinding> fetchServiceBindings() {
+        Map<String, ServiceBindingProperties> serviceBindingsProperties =
+                SapServiceBindingsPropertiesAccessor.getServiceBindingsProperties();
+        return serviceBindingsProperties
+                .entrySet()
+                .stream()
+                .map(entry -> toServiceBinding(entry.getKey(), entry.getValue()))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    @Nullable
+    private ServiceBinding toServiceBinding(
+            @Nonnull String serviceName,
+            @Nonnull final ServiceBindingProperties serviceBindingProperties) {
+        if (validateServiceBindingProperties(serviceBindingProperties)) {
+            return DefaultServiceBinding
+                    .builder()
+                    .copy(Collections.emptyMap())
+                    .withName(serviceBindingProperties.getName())
+                    .withServiceName(serviceName)
+                    .withServicePlan(serviceBindingProperties.getPlan())
+                    .withTags(
+                            Arrays.asList(serviceBindingProperties.getTags())
+                    )
+                    .withCredentials(
+                            serviceBindingProperties.getCredentials())
+                    .build();
+        } else {
+            return null;
+        }
+    }
+
+    private boolean validateServiceBindingProperties(@Nonnull final ServiceBindingProperties serviceBindingProperties) {
+        boolean isValid = true;
+        if (serviceBindingProperties.getName() == null || serviceBindingProperties.getName().isEmpty()) {
+            logger.error("Service binding properties with no name detected.");
+            isValid = false;
+        }
+
+        if (serviceBindingProperties.getLabel() == null || serviceBindingProperties.getLabel().isEmpty()) {
+            logger.error("Service binding properties of {} has no label.", serviceBindingProperties.getName());
+            isValid = false;
+        }
+
+        if (serviceBindingProperties.getTags() == null || serviceBindingProperties.getTags().length == 0) {
```

Comment:
Why are tags mandatory? Could default to an empty tag list..

Reply:
I had the impression without tags it didn't work. I'll verify it again, as I'm not sure...